src/forms/inputs/cloudinary-file-input.js

```diff
@@ -53,25 +52,40 @@ const defaultProps = {
   onUploadFailure: noop,
 }
 
-function CloudinaryFileInput ({ 
+function mapCloudinaryResponse (file, response) {
+  return compose(
+    set('url', response.url),
+    set('meta.cloudinary', response)
+  )(file)
+}
+
+function CloudinaryFileInput ({
   input,
   className,
   onUploadFailure,
   onUploadSuccess,
-  upload, 
-  uploadStatus, 
+  upload,
+  uploadStatus,
+  fileInput: FileInput = DefaultFileInput,
   ...rest 
 }) {
-  const { onChange } = input
   return (
     <FileInput
-      input={{ ...input, onChange: noop }}
-      onLoad={ (fileData, file) => upload(fileData, file)
-        .then((res) => {
-          onChange(res.url)
-          return onUploadSuccess(res)
-        }, (err) => onUploadFailure(err))
-      }
+      input={ input }
+      onRead={async (files) => {
+        try {
+          const uploadFilePromises = files.map(async (file) => {
+            const cloudinaryRes = await upload(file.url, file)
+            return mapCloudinaryResponse(file, cloudinaryRes)
+          })
+          
+          const uploadedFiles = await Promise.all(uploadFilePromises)
+          onUploadSuccess(uploadedFiles)
+          return uploadedFiles
+        } catch (e) {
```

Comment:
You'll need to rearrange this `catch` to match the previous behavior (don't catch exceptions from `onUploadSuccess`)